aws/resource_aws_iam_group_policy.go

```diff
@@ -105,17 +117,22 @@ func resourceAwsIamGroupPolicyRead(d *schema.ResourceData, meta interface{}) err
 		return err
 	}
 
-	d.Set("group", group)
-	d.Set("name", name)
-	d.Set("policy", policy)
-
-	return nil
+	if err := d.Set("policy", policy); err != nil {
+		return err
+	}
+	if err := d.Set("name", name); err != nil {
+		return err
```

Comment:
Same here regarding error context. 👍 

```suggestion
		return fmt.Errorf("error setting name: %s", err)
```

Reply:
Done.